### PR TITLE
Adding loader for TAU SSE 2019 dataset

### PR DESCRIPTION
**Pull request by iranroman**
### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the total duration of the dataset in hours, and (optionally) also list the number of individual files
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of audio included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `soundata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in soundata, e.g. `soundata/my_dataset.py`
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/soundata.rst` and `docs/source/quick_reference.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

### CODE REVIEW

**Comment by iranroman** [comment]
I've tested that the this loader will work with the latest changes in `master`. Ready to be reviewed and merged.

File: docs/source/table.rst [hunk 1]
```diff
@@ -54,6 +54,14 @@
      - .. image:: https://licensebuttons.net/l/by/4.0/80x15.png
           :target: https://creativecommons.org/licenses/by/4.0
 
+   * - TAU Spatial Sound Events 2019
```
Comment (magdalenafuentes) [inline docs/source/table.rst]: Maybe break this into two lines for rendering

File: docs/source/table.rst [hunk 2]
```diff
@@ -54,6 +54,14 @@
...
+     - - audio: ✅
+       - annotations: ✅
+     - :ref:`events`
+     - 500
+     - 8.3
+     - :tau_sse_2019:`\ `
```
Comment (magdalenafuentes) [inline docs/source/table.rst:65]: Is this defined anywhere? is not rendering properly in the docs

File: soundata/datasets/tau_sse_2019.py [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+    *Created By:*
+        Sharath Adavanne; Archontis Politis; Tuomas Virtanen
+        Audio Research Group, Tampere University. Version 2
+
+    *Description:*
+        Recordings with stationary point sources (events) from multiple sound classes.
+        Up to two temporally overlaping sound events. 
+        Recordings of identical scenes are available in both 1st-order ambisonics and corresponding four-channel tetrahedral microphone format.
+        Recordings can happen in one of five different rooms.
+        The sound classes are the 11 different ones from the [DCASE 2016 challenge task 2](http://dcase.community/challenge2016/task-sound-event-detection-in-synthetic-audio). Each class has 20 different examples.
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py]: Link is not rendering properly

File: soundata/datasets/tau_sse_2019.py [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+import csv
+import jams
+import glob
+import json
+
+from soundata import download_utils
+from soundata import jams_utils
+from soundata import core
+from soundata import annotations
+from soundata import io
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py]: all of this in one line

File: soundata/datasets/tau_sse_2019.py [hunk 5]
```diff
@@ -0,0 +1,397 @@
...
+    title = "A Multi-room Reverberant Dataset for Sound Event Localization and Uetection",
+    year = "2019",
+    booktitle = "Submitted to Detection and Classification of Acoustic Scenes and Events 2019 Workshop (DCASE2019)",
+    abstract = "This paper presents the sound event localization and detection (SELD) task setup for the DCASE 2019 challenge. The goal of the SELD task is to detect the temporal activities of a known set of sound event classes, and further localize them in space when active. As part of the challenge, a synthesized dataset with each sound event associated with a spatial coordinate represented using azimuth and elevation angles is provided. These sound events are spatialized using real-life impulse responses collected at multiple spatial coordinates in five different rooms with varying dimensions and material properties. A baseline SELD method employing a convolutional recurrent neural network is used to generate benchmark scores for this reverberant dataset. The benchmark scores are obtained using the recommended cross-validation setup.",
+    url = "https://arxiv.org/abs/1905.08546"
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py]: Is it published in any venue instead?

File: soundata/datasets/tau_sse_2019.py [hunk 6]
```diff
@@ -0,0 +1,397 @@
...
+    return events_data
+
+
+def validate_locations(locations):
+    """Validate if TAU SSE 2019 locations are well-formed.
+
+    If locations is None, validation passes automatically
+
+    Args:
+        intervals (np.ndarray): (n x 3) array
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py]: locations?

File: soundata/datasets/tau_sse_2019.py [hunk 7]
```diff
@@ -0,0 +1,397 @@
...
+
+    Raises:
+        ValueError: if intervals have an invalid shape or
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py]: locations?

File: soundata/datasets/tau_sse_2019.py [hunk 8]
```diff
@@ -0,0 +1,397 @@
...
+                have cartesian coordinate values outside the expected ranges.
+    """
+    if locations is None:
+        return
+
+    # validate that locations have the correct shape
+    locations_shape = np.shape(locations)
+    if len(locations_shape) != 2 or locations_shape[1] != 3:
+        raise ValueError(
+            f"Locations should be arrays with three columns, but array has {locations_shape}"
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py]: array has `shape` {locations_shape}

File: soundata/datasets/tau_sse_2019.py [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+        )
+
+    # validate that values are within expected ranges
+    if (np.abs(locations[:, 0]) > 90).any():
+        raise ValueError(f"Elevation values should have magnitude less than 90")
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py:351]: less -> bigger

File: soundata/datasets/tau_sse_2019.py [hunk 10]
```diff
@@ -0,0 +1,397 @@
...
+    if (np.abs(locations[:, 1]) > 180).any():
+        raise ValueError(f"Azimuth values should have magnitude less than 180")
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py:353]: less -> bigger

File: soundata/datasets/tau_sse_2019.py [hunk 11]
```diff
@@ -0,0 +1,397 @@
...
+
+
+class Clip(core.Clip):
+    """TAU SSE 2019 Clip class
+
+    Args:
+        clip_id (str): id of the clip
+
+    Attributes:
+        spatial_events (SpatialEvents): sound events with start time, end time, elevation, azimuth, distance, label and confidence.
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py:195]: I wonder if there's a way we can make this one clearer (e.g. `soundata.annotations.Events` is clear that is a class in annotations. Thoughts?

File: soundata/datasets/tau_sse_2019.py [hunk 12]
```diff
@@ -0,0 +1,397 @@
...
+        """The clip's spatial events
+
+        Returns:
+            * SpatialEvents class with attributes
+                * intervals (np.ndarray): (n x 2) array of intervals
+                    (as floats) in seconds in the form [start_time, end_time]
+                    with positive time stamps and end_time >= start_time.
+                * elevations (np.ndarray): (n,) aarray of elevations
+                * azimuths (np.ndarray): (n,) aarray of azimuths
+                * distances (np.ndarray): (n,) aarray of distances
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py]: aarray -> array

File: soundata/datasets/tau_sse_2019.py [hunk 13]
```diff
@@ -0,0 +1,397 @@
...
+        return load_spatialevents(self.csv_path)
+
+    def to_jams(self):
+        """Get the clip's data in jams format
+
+        Returns:
+            jams.JAMS: the clip's data in jams format
+
+        """
+        return jams_utils.jams_converter(audio_path=self.audio_path)
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_sse_2019.py]: maybe we can also return some metadata?

**Review by magdalenafuentes — COMMENTED** [review]
@iranroman this looks great! I left very minor comments and is good to go, good job!

File: docs/source/table.rst (hunk 2 above)
Comment (iranroman) [reply docs/source/table.rst:65]: It's not. The authors released this dataset in 2019 with their own copyright. See here: https://zenodo.org/record/2580091

File: soundata/datasets/tau_sse_2019.py (hunk 3 above)
Comment (iranroman) [reply soundata/datasets/tau_sse_2019.py]: The markdown link seems to work. Here is written in the same way:
[DCASE 2016 challenge task 2](http://dcase.community/challenge2016/task-sound-event-detection-in-synthetic-audio)

File: soundata/datasets/tau_sse_2019.py (hunk 4 above)
Comment (iranroman) [reply soundata/datasets/tau_sse_2019.py]: All other loaders do it this way.

File: soundata/datasets/tau_sse_2019.py (hunk 11 above)
Comment (iranroman) [reply soundata/datasets/tau_sse_2019.py:195]: I was wondering the same thing. The ideal solution would be the inclusion of this class in `annotations.py`.

File: soundata/datasets/tau_sse_2019.py (hunk 5 above)
Comment (iranroman) [reply soundata/datasets/tau_sse_2019.py]: Found the DCASE workshop version and included that instead.

File: soundata/datasets/tau_sse_2019.py (hunk 9 above)
Comment (iranroman) [reply soundata/datasets/tau_sse_2019.py:351]: I think it's fine as it is. No?

File: soundata/datasets/tau_sse_2019.py (hunk 10 above)
Comment (iranroman) [reply soundata/datasets/tau_sse_2019.py:353]: I think it's fine as it is. No?

File: docs/source/table.rst (hunk 2 above)
Comment (magdalenafuentes) [reply docs/source/table.rst:65]: I meant if the link is defined anywhere. There are other cases where loaders have custom licenses, and they display the word custom and a link to the license. See [this example](https://github.com/soundata/soundata/blob/master/docs/conf.py#L59).

File: soundata/datasets/tau_sse_2019.py (hunk 3 above)
Comment (magdalenafuentes) [reply soundata/datasets/tau_sse_2019.py]: Is not rendering properly though, see [the docs of the PR](https://soundata--74.org.readthedocs.build/en/74/source/soundata.html#module-soundata.datasets.tau_sse_2019)

File: soundata/datasets/tau_sse_2019.py (hunk 4 above)
Comment (magdalenafuentes) [reply soundata/datasets/tau_sse_2019.py]: That's not true, only 2/10 do it that way, and is better to have it in one line

File: soundata/datasets/tau_sse_2019.py (hunk 5 above)
Comment (magdalenafuentes) [reply soundata/datasets/tau_sse_2019.py]: Great, thanks!

File: soundata/datasets/tau_sse_2019.py (hunk 9 above)
Comment (magdalenafuentes) [reply soundata/datasets/tau_sse_2019.py:351]: Mmmm yeah I think you're right! I was having trouble parsing it... maybe say that angles are not the interval [-90, 90] would make it clearer?

File: soundata/datasets/tau_sse_2019.py (hunk 11 above)
Comment (magdalenafuentes) [reply soundata/datasets/tau_sse_2019.py:195]: Yeah... I think I agree with you

File: soundata/datasets/tau_sse_2019.py (hunk 3 above)
Comment (iranroman) [reply soundata/datasets/tau_sse_2019.py]: Interesting. No idea. Maybe there's an issue with the doc's markdown source renderer?

**Review by magdalenafuentes — APPROVED** [review]

**Comment by magdalenafuentes** [comment]
Great one @iranroman! Merging this!

**Pull request merged**